pkg/service/presentation/service.go

```diff
@@ -227,6 +231,27 @@ func (s Service) CreateSubmission(ctx context.Context, request model.CreateSubmi
 	}, nil
 }
 
+// verifyTokenFromDID verifies the token from the did by first resolving the did, next extracting the public key,
+// and then verifying the token
+func (s Service) verifyTokenFromDID(ctx context.Context, did string, token keyaccess.JWT) error {
```

Comment:
This function is repeated. Was this intentional?

Reply:
no - will remove one

Reply:
this is intentional. it is because one is in the presentation service and another in the manifest service. each service has its own dependencies and separate impls, but has two options to verify a token:
1. call the new method in the DID service - `GetKeyFromDID` and verify the token
2. use the resolver interface as it's doing

Reply:
an alternative could be to add a new method to the DID service that does this verification

Reply:
actually now that the resolver interface is in the SDK I can undo this